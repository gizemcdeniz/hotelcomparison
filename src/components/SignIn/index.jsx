import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";
import SignInForm from "../SignInForm";
import Boring from "../../images/boring.png";
import googleIcon from "../../images/icons/googleIcon.svg";
import twitterIcon from "../../images/icons/twitterIcon.svg";
import facebookIcon from "../../images/icons/facebookIcon.svg";
import { setFirebaseProvider } from "../../utils/authHelpers";
import { NavLink } from "react-router-dom";
import firebase from "../../firebaseConfig";

import "./index.scss";

export default function SignIn() {
  const { t } = useTranslation();
  const history = useHistory();

  const signInFailed = (error) => {
    alert(error);
  };

  const handleSignIn = React.useCallback(
    async (data) => {
      const { email, password } = data;

      try {
        await firebase.auth().signInWithEmailAndPassword(email, password);
        history.push("/");
      } catch (error) {
        if (
          error.code === "auth/wrong-password" ||
          error.code === "auth/user-not-found"
        ) {
          return error.code;
        }
        signInFailed(error);
      }
      return "succeed";
    },
    [history]
  );

  const handleSignUpWithProvider = React.useCallback(
    async (providerName) => {
      const firebaseProvider = setFirebaseProvider(providerName);

      firebase
        .auth()
        .signInWithPopup(firebaseProvider)
        .then(function (result) {
          history.push("/");
        })
        .catch(function (error) {
          // Handle errors here.
          signInFailed(error);
        });
    },
    [history]
  );

  return (
    <Container fluid="md" className="signInSection">
      <Row>
        <Col className="signInImg" md={6} xs={12}>
          <img
            src={Boring}
            alt="Women Group Hugging Together"
          />
        </Col>
        <Col className="signInForm" md={6} xs={12}>
          <h2>{t("signIn.header")}</h2>
          <div>
            <span className="hyphen" />
            <h4>{t("signIn.signInWith")}</h4>
          </div>
          <ul className="signInIcons">
            <li className="signInIcon">
              <a href="#/" onClick={() => handleSignUpWithProvider("twitter")}>
                <img src={twitterIcon} alt="Twitter icon" />
              </a>
            </li>
            <li className="signInIcon">
              <a href="#/" onClick={() => handleSignUpWithProvider("facebook")}>
                <img src={facebookIcon} alt="Facebook icon" />
              </a>
            </li>
            <li className="signInIcon">
              <a href="#/" onClick={() => handleSignUpWithProvider("google")}>
                <img src={googleIcon} alt="Google icon" />
              </a>
            </li>
          </ul>
          <p className="text-center">
            <hr className="divider" />
            <span className="dividerContent">{t("signIn.or")}</span>
          </p>
          <SignInForm submit={handleSignIn} />

          <p>
            {t("signIn.newHere")}{" "}
            <NavLink to="/signup">{t("signIn.createAccount")}</NavLink>
          </p>
        </Col>
      </Row>
    </Container>
  );
}
