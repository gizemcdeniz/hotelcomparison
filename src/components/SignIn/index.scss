@import "../../style/variables";
@import "../../style/buttons";
@import "../../style/headings";
@import "../../style/bootstrapImports";
@import "../../style/shapes";

.signInSection {
  /* All section style */
  width: 100%;
  margin-top: 4rem;
  margin-bottom: 4rem;

  .signInImg {
    background-color: $lightColor;
    margin: auto 0;

    img {
      width: 100%;
    }
  }

  .signInForm {
    background-color: $lightColor;
  }

  h2 {
    @include mainHeading;
    font-size: 2.8rem;
    color: $primaryColor;
    margin-bottom: 4rem;
  }

  h4 {
    display: inline-block;
    align-content: space-between;
    font-size: 1.25rem;
    color: $primaryColor;
    margin: 0 0.8rem 2rem;
  }

  .hyphen {
    display: inline-block;
    border-top: 0.4rem solid #fe8f7c;
    border-radius: $borderRadius;
    width: 3.5rem;
    height: 0;
    margin: auto 0;
  }

  a {
    color: $primaryColor;

    &:hover {
      text-decoration: none;
      color: $warningColor;
    }
  }

  hr.divider {
    @include sectionDividerLine;
  }

  .dividerContent {
    content: "Or";
    display: inline-block;
    text-align: center;
    margin: 0 auto;
    position: relative;
    top: -1.4rem;
    padding: 0 1rem;
    background: $lightColor;
  }

  /* Icons Style */
  .signInIcons {
    box-sizing: border-box;
    display: flex;
    justify-content: space-around;
    margin-bottom: 2rem;
    padding: 0;
  }

  .signInIcon {
    list-style: none;
    background-color: $successColor;
    border-radius: 50px;
    padding: 0.6rem;
    margin: 1rem 0;

    &:hover {
      background-color: $primaryColor;
    }
  }

  .signInIcon img {
    width: 2rem;
    height: 2rem;
  }
}
