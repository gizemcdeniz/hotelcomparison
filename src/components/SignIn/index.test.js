import React from "react";
import renderer from "react-test-renderer";
import SignIn from "./index";
import { BrowserRouter as Router } from "react-router-dom";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <SignIn />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
