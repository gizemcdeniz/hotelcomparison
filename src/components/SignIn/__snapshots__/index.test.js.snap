// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="signInSection container-md"
>
  <div
    className="row"
  >
    <div
      className="signInImg col-md-6 col-12"
    >
      <img
        alt="Women Group Hugging Together"
        src="womenGroupHuggingTogether.png"
      />
    </div>
    <div
      className="signInForm col-md-6 col-12"
    >
      <h2>
        signIn.header
      </h2>
      <div>
        <span
          className="hyphen"
        />
        <h4>
          signIn.signInWith
        </h4>
      </div>
      <ul
        className="signInIcons"
      >
        <li
          className="signInIcon"
        >
          <a
            href="#/"
            onClick={[Function]}
          >
            <img
              alt="Twitter icon"
              src="twitterIcon.svg"
            />
          </a>
        </li>
        <li
          className="signInIcon"
        >
          <a
            href="#/"
            onClick={[Function]}
          >
            <img
              alt="Facebook icon"
              src="facebookIcon.svg"
            />
          </a>
        </li>
        <li
          className="signInIcon"
        >
          <a
            href="#/"
            onClick={[Function]}
          >
            <img
              alt="Google icon"
              src="googleIcon.svg"
            />
          </a>
        </li>
      </ul>
      <p
        className="text-center"
      >
        <hr
          className="divider"
        />
        <span
          className="dividerContent"
        >
          signIn.or
        </span>
      </p>
      <div
        className="signInForm"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="formInput"
          >
            <input
              aria-label="email"
              name="email"
              placeholder="signIn.form.yourEmail"
              type="email"
            />
          </div>
          <div
            className="formInput"
          >
            <input
              aria-label="password"
              name="password"
              placeholder="signIn.form.yourPassword"
              type="password"
            />
          </div>
          <div
            className="formInput"
          >
            <div
              className="rememberMe"
            >
              <div>
                <input
                  id="remembered"
                  name="remembered"
                  type="checkbox"
                  value="toBeRemembered"
                />
              </div>
              <div>
                <label
                  htmlFor="remembered"
                >
                  signIn.form.rememberMe
                </label>
              </div>
            </div>
          </div>
          <button
            className="submitBtn"
            type="submit"
          >
            signIn.form.signIn
          </button>
        </form>
      </div>
      <p>
        signIn.newHere
         
        <a
          aria-current={null}
          href="/signup"
          onClick={[Function]}
        >
          signIn.createAccount
        </a>
      </p>
    </div>
  </div>
</div>
`;
